core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
Whatever you call, this is something non-standard and depends on the internal decisions used for hashmap. I can show one at least one very different implementation [in Go](https://golang.org/src/runtime/alg.go#L101). And see how they document the behavior that deviates from IEEE-754. Concise, to the point, and brief. See how they don't brush it off by just saying "safe".

I really don't understand your resistance for not documenting this. You could've written a single line about what is expected and why rather than writing paragraphs arguing about it.